converter.go

```diff
@@ -952,22 +971,34 @@ func higherPrecisionEnabled(ctx context.Context) bool {
 	return ok && d
 }
 
-func arrowToRecord(record arrow.Record, pool memory.Allocator, rowType []execResponseRowType, loc *time.Location) (arrow.Record, error) {
-	s, err := recordToSchema(record.Schema(), rowType, loc)
+func originalTimestampEnabled(ctx context.Context) bool {
+	v := ctx.Value(enableOriginalTimestamp)
+	if v == nil {
+		return false
+	}
+	d, ok := v.(bool)
+	return ok && d
+}
+
+func arrowToRecord(ctx context.Context, record arrow.Record, pool memory.Allocator, rowType []execResponseRowType, loc *time.Location) (arrow.Record, error) {
+	useOriginalTimestamp := originalTimestampEnabled(ctx)
+
+	s, err := recordToSchema(record.Schema(), rowType, loc, useOriginalTimestamp)
 	if err != nil {
 		return nil, err
 	}
 
 	var cols []arrow.Array
 	numRows := record.NumRows()
-	ctx := compute.WithAllocator(context.Background(), pool)
+	ctxAlloc := compute.WithAllocator(ctx, pool)
```

Comment:
I'm just wondering about one thing. What if someone specify their own allocator in the context? For instance when we have allocator with memory checking for tests? Worked so far, so maybe this is not an issue...

Reply:
Worked so far, let's leave it as it is.